### PR TITLE
feat: specify the default viewport

### PR DESCRIPTION
**Pull request by JoelEinbinder**
Removes the undocumented `appMode`, adds a public option `defaultViewport`

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -359,6 +359,13 @@ puppeteer.launch().then(async browser => {
 - `options` <[Object]>
   - `browserWSEndpoint` <[string]> a [browser websocket endpoint](#browserwsendpoint) to connect to.
   - `ignoreHTTPSErrors` <[boolean]> Whether to ignore HTTPS errors during navigation. Defaults to `false`.
+  - `defaultViewport` <?[Object]> . Defaults to an 800x600 viewport. `null` disables the default viewport.
```
Comment (vsemozhetbyt) [inline docs/api.md]: Nit: unneeded spaces and period in `<?[Object]> . Defaults`?

File: docs/api.md [hunk 2]
```diff
@@ -383,6 +390,13 @@ This methods attaches Puppeteer to an existing Chromium instance.
   - `headless` <[boolean]> Whether to run browser in [headless mode](https://developers.google.com/web/updates/2017/04/headless-chrome). Defaults to `true` unless the `devtools` option is `true`.
   - `executablePath` <[string]> Path to a Chromium or Chrome executable to run instead of the bundled Chromium. If `executablePath` is a relative path, then it is resolved relative to [current working directory](https://nodejs.org/api/process.html#process_process_cwd).
   - `slowMo` <[number]> Slows down Puppeteer operations by the specified amount of milliseconds. Useful so that you can see what is going on.
+  - `defaultViewport` <?[Object]> . Defaults to an 800x600 viewport. `null` disables the default viewport.
```
Comment (vsemozhetbyt) [inline docs/api.md]: Ditto.

File: docs/api.md (hunk 1 above)
Comment (JoelEinbinder) [reply docs/api.md]: Oops, accidentally deleted the sentence that was there. Good catch!

**Review by aslushnikov — APPROVED** [review]

File: lib/Launcher.js [hunk 3]
```diff
@@ -155,17 +150,19 @@ class Launcher {
     /** @type {?Connection} */
     let connection = null;
     try {
-      const connectionDelay = options.slowMo || 0;
+      const {
+        ignoreHTTPSErrors = false,
+        defaultViewport = {width: 800, height: 600},
+        slowMo = 0
+      } = options;
```
Comment (Palgie) [inline lib/Launcher.js:157]: Will this not cause timeout to be `null` below and always default to 30000?

Comment (Palgie) [reply lib/Launcher.js:157]: Getting issues with third parties using puppeteer where users are reporting not being able to change timeout anymore.

**Comment by Palgie** [comment]
@aslushnikov @vsemozhetbyt

File: lib/Launcher.js (hunk 3 above)
Comment (JoelEinbinder) [reply lib/Launcher.js:157]: The code for setting the timeout is two lines below this, and remains unchanged. Can you file an issue? I don't think this patch is the cause.

Comment (cbravo) [reply lib/Launcher.js:157]: @Palgie this code above is simply pulling values OUT of the options object using es6 destructuring . So it is not the cause of the timeout value not being set as you described [here](https://github.com/chrisvfritz/prerender-spa-plugin/issues/222)

**Pull request merged**